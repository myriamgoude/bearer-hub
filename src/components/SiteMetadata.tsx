/*
  NB: This component sets default metadata individual templates/pages can override this
*/

import * as React from 'react'
import { Helmet } from 'react-helmet'
import { graphql, StaticQuery } from 'gatsby'

interface ISiteMetadataProps {
  pathname: string
}

const query = graphql`
  query SiteMetadata {
    site {
      siteMetadata {
        siteUrl
        title
        twitter
        description
      }
    }
  }
`
interface IDefaultMetaProps {
  title: string
  canonical: string
  twitter: string
  image: string
  description: string
}

// export this so we can test it with a shallow mount
// NB: Full mount is required to mock the query but helmet will not return anything
// in a full mount as it edits the header!
export const DefaultMeta: React.SFC<IDefaultMetaProps> = ({ title, canonical, image, twitter, description }) => (
  <Helmet defaultTitle={title} titleTemplate={`%s | ${title}`}>
    <html lang="en" />
    <link rel="canonical" href={canonical} />

    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:title" content={title} />
    <meta name="twitter:description" content={description} />
    <meta name="twitter:image" content={image} />
    <meta name="twitter:site" content={twitter} />

    <meta property="og:site_name" content={title} />
    <meta property="og:title" content={title} />
    <meta property="og:description" content={description} />
    <meta property="og:url" content={canonical} />
    <meta property="og:locale" content="en" />
    <meta property="og:type" content="website" />
    <meta property="og:image" content={image} />
    <meta property="og:image:width" content="1200" />
    <meta property="og:image:height" content="630" />
  </Helmet>
)

const SiteMetadata = ({ pathname }: ISiteMetadataProps) => (
  <StaticQuery
    query={query}
    render={({
      site: {
        siteMetadata: { siteUrl, title, twitter, description }
      }
    }) => {
      const meta = {
        siteUrl,
        title,
        twitter,
        description,
        image: 'https://www.bearer.sh/static/media/slack-notification-full-size.29ef4908.png',
        canonical: `${siteUrl}${pathname}`
      }
      return <DefaultMeta {...meta} />
    }}
  />
)

export default SiteMetadata
