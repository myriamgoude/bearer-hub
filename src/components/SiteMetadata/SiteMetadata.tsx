/*
  NB: This component sets default metadata individual templates/pages can override this
*/

import * as React from 'react'
import { Helmet } from 'react-helmet'
import { graphql, StaticQuery } from 'gatsby'

import favicon_32x32 from '../../../static/favicon-32x32.png'
import favicon_16x16 from '../../../static/favicon-16x16.png'
import apple_touch_icon from '../../../static/apple-touch-icon.png'
import safari_svg from '../../../static/safari-pinned-tab.svg'

import facebookDefault_1x from '../../../static/FB-default@1x.png'

interface ISiteMetadataProps {
  pathname: string
}

const query = graphql`
  query SiteMetadata {
    site {
      siteMetadata {
        siteUrl
        title
        twitter
        description
      }
    }
  }
`

interface IDefaultMetaProps {
  title: string
  canonical: string
  twitter: string
  image: string
  alt: string
  description: string
}

// export this so we can test it with a shallow mount
// NB: Full mount is required to mock the query but helmet will not return anything
// in a full mount as it edits the header!
export const DefaultMeta: React.SFC<IDefaultMetaProps> = ({ title, canonical, image, alt, twitter, description }) => (
  <Helmet defaultTitle={title} titleTemplate={`%s | ${title}`}>
    <html lang="en" />
    <link rel="canonical" href={canonical} />
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:title" content={title} />
    <meta name="twitter:description" content={description} />
    <meta name="twitter:image" content={image} />
    <meta name="twitter:image:alt" content={alt ? alt : ''} />
    <meta name="twitter:site" content={twitter} />
    <meta property="og:site_name" content={title} />
    <meta property="og:title" content={title} />
    <meta property="og:description" content={description} />
    <meta property="og:url" content={canonical} />
    <meta property="og:locale" content="en" />
    <meta property="og:type" content="website" />
    <meta property="og:image" content={image} />
    <meta property="og:image:alt" content={alt ? alt : ''} />
    <meta property="og:image:width" content="1200" />
    <meta property="og:image:height" content="630" />

    <link rel="icon" sizes="16x16" type="image/png" href={favicon_16x16} />
    <link rel="icon" sizes="32x32" type="image/png" href={favicon_32x32} />
    <link rel="icon" sizes="180x180" type="image/png" href={apple_touch_icon} />
    <link rel="mask-icon" href={safari_svg} />

    <meta name="theme-color" content="#da532c" />
  </Helmet>
)

const SiteMetadata = ({ pathname }: ISiteMetadataProps) => (
  <StaticQuery
    query={query}
    render={({
      site: {
        siteMetadata: { siteUrl, title, twitter, description }
      }
    }) => {
      const meta = {
        siteUrl,
        title,
        twitter,
        description,
        image: `${facebookDefault_1x}`,
        alt: 'Bearer logo with slogan "Developer Platform for API Integration"',
        canonical: `${siteUrl}${pathname}`
      }
      return <DefaultMeta {...meta} />
    }}
  />
)

export default SiteMetadata
