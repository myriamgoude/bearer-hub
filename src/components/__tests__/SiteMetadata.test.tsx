import React from 'react'
import * as ShallowRenderer from 'react-test-renderer/shallow'
import { DefaultMeta } from '../SiteMetadata/SiteMetadata'

describe('SiteMetadata', () => {
  const props = {
    title: 'Bearer',
    canonical: 'https://example.com/my-page',
    twitter: '@bearersh',
    image: 'https://example.com/my-image.jpg',
    description: 'This is an example description'
  }
  it('DefaultMeta renders expected metadata', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<DefaultMeta {...props} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
