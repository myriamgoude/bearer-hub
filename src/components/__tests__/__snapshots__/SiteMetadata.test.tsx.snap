// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteMetadata DefaultMeta renders expected metadata 1`] = `
<HelmetWrapper
  defaultTitle="Bearer"
  defer={true}
  encodeSpecialCharacters={true}
  titleTemplate="%s"
>
  <html
    lang="en"
  />
  <link
    href="https://example.com/my-page"
    rel="canonical"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="Bearer"
    name="twitter:title"
  />
  <meta
    content="This is an example description"
    name="twitter:description"
  />
  <meta
    content="https://example.com/my-image.jpg"
    name="twitter:image"
  />
  <meta
    content="An example description of my image"
    name="twitter:image:alt"
  />
  <meta
    content="@bearersh"
    name="twitter:site"
  />
  <meta
    content="Bearer"
    property="og:site_name"
  />
  <meta
    content="Bearer"
    property="og:title"
  />
  <meta
    content="This is an example description"
    property="og:description"
  />
  <meta
    content="https://example.com/my-page"
    property="og:url"
  />
  <meta
    content="en"
    property="og:locale"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="https://example.com/my-image.jpg"
    property="og:image"
  />
  <meta
    content="An example description of my image"
    property="og:image:alt"
  />
  <meta
    content="1200"
    property="og:image:width"
  />
  <meta
    content="630"
    property="og:image:height"
  />
  <link
    href="test-file-stub"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="test-file-stub"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="test-file-stub"
    rel="icon"
    sizes="180x180"
    type="image/png"
  />
  <link
    href="test-file-stub"
    rel="mask-icon"
  />
  <meta
    content="#da532c"
    name="theme-color"
  />
</HelmetWrapper>
`;
